lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -1540,12 +1607,398 @@ void amf_app_handle_initial_context_setup_rsp(
+        pdu_session_resource_modification_t3591_handler, id);
+
+  } else {
+    OAILOG_WARNING(LOG_AMF_APP, "NAS encode failed \n");
+    bdestroy_wrapper(&buffer);
+  }
+
+  return rc;
+}
+void amf_app_fill_create_new_tft(create_new_tft_t* new_tft, QOSRule* qos_rule) {
+  for (int i = 0; i < qos_rule->no_of_pkt_filters; i++) {
+    NewQOSRulePktFilter new_qos_rule_pkt_filter = {};
+    packet_filter_t* pkt_filter = NULL;
+    uint16_t pkt_filter_len = 0;
+    pkt_filter = &new_tft[i];
+    memset(&new_qos_rule_pkt_filter, 0, sizeof(NewQOSRulePktFilter));
+    new_qos_rule_pkt_filter.spare = 0x0;
+    new_qos_rule_pkt_filter.pkt_filter_dir = pkt_filter->direction;
+    new_qos_rule_pkt_filter.pkt_filter_id = pkt_filter->identifier;
+    uint16_t flag = TRAFFIC_FLOW_TEMPLATE_IPV4_REMOTE_ADDR_FLAG;
+    while (flag <= TRAFFIC_FLOW_TEMPLATE_FLOW_LABEL_FLAG) {
+      switch (pkt_filter->packetfiltercontents.flags & flag) {
+        case TRAFFIC_FLOW_TEMPLATE_IPV4_REMOTE_ADDR_FLAG: {
+          /*
+           * IPv4 remote address type
+           */
+          new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+              TRAFFIC_FLOW_TEMPLATE_IPV4_REMOTE_ADDR;
+          pkt_filter_len++;
+
+          for (int j = 0; j < TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE; j++) {
+            new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+                pkt_filter->packetfiltercontents.ipv4remoteaddr[j].addr;
+            new_qos_rule_pkt_filter
+                .contents[pkt_filter_len +
+                          TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE] =
+                pkt_filter->packetfiltercontents.ipv4remoteaddr[j].mask;
+            pkt_filter_len++;
+          }
+
+          pkt_filter_len += TRAFFIC_FLOW_TEMPLATE_IPV4_ADDR_SIZE;
+        } break;
+        case TRAFFIC_FLOW_TEMPLATE_SINGLE_REMOTE_PORT_FLAG: {
+          /*
+           * Remote port type
+           */
+          new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+              TRAFFIC_FLOW_TEMPLATE_SINGLE_REMOTE_PORT;
+          pkt_filter_len++;
+
+          new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+              (0xFF00 & pkt_filter->packetfiltercontents.singleremoteport) >> 8;
+          pkt_filter_len++;
+          new_qos_rule_pkt_filter.contents[pkt_filter_len] =
+              (0x00FF & pkt_filter->packetfiltercontents.singleremoteport);
+          pkt_filter_len++;
+        } break;
+        default: {
+        }
+      }
+      flag = flag << 1;
+    }
+    new_qos_rule_pkt_filter.len = pkt_filter_len;
+    memcpy(&qos_rule->new_qos_rule_pkt_filter[0], &new_qos_rule_pkt_filter,
+           sizeof(NewQOSRulePktFilter));
+    qos_rule->len += new_qos_rule_pkt_filter.len + 2;
+  }
+}
+
+void amf_app_fill_delete_packet_filter(
+    delete_packet_filter_t* delete_pkt_filter, QOSRule* qos_rule) {
+  NewQOSRulePktFilter new_qos_rule_pkt_filter = {};
+  delete_packet_filter_t* pkt_filter = NULL;
+  for (int i = 0; i < qos_rule->no_of_pkt_filters; i++) {
```

Comment:
General comment: don't use the index as "i", use appropriate name to indicate through which list it's looping through